pkg/kudoctl/util/kudo/kudo.go

```diff
@@ -42,44 +43,27 @@ func NewClient(kubeConfigPath string, requestTimeout int64) (*Client, error) {
 	// set default configs
 	config.Timeout = time.Duration(requestTimeout) * time.Second
 
-	// create the clientset
-	kudoClientset, err := versioned.NewForConfig(config)
+	kubeClient, err := kube.GetKubeClient(kubeConfigPath)
 	if err != nil {
-		return nil, err
+		return nil, clog.Errorf("could not get Kubernetes client: %s", err)
 	}
 
-	// use the apiextensions clientset to check for the existence of KUDO CRDs in the cluster
-	extensionsClientset, err := extensionsclient.NewForConfig(config)
+	err = kudoinit.CRDs().ValidateInstallation(kubeClient)
 	if err != nil {
-		return nil, err
-	}
-
-	_, err = extensionsClientset.CustomResourceDefinitions().Get("operators.kudo.dev", v1.GetOptions{})
-	if err != nil {
-		// timeout is not a wrappable error, timeout is an underlying issue that is NOT CRD specific, there is no value in wrapping or converting as well.
-		// best to provide the actual error for proper reporting.
+		// see above
 		if os.IsTimeout(err) {
 			return nil, err
 		}
-		return nil, fmt.Errorf("operators crd: %w", err)
-	}
-
-	_, err = extensionsClientset.CustomResourceDefinitions().Get("operatorversions.kudo.dev", v1.GetOptions{})
-	if err != nil {
-		// timeout details above for first CRD
-		if os.IsTimeout(err) {
-			return nil, err
+		clog.V(0).Printf("Cluster CRDS are not set up correctly. Do you need to run kudo init?")
+		if validateInstall {
+			return nil, fmt.Errorf("CRDs invalid: %v", err)
 		}
-		return nil, fmt.Errorf("operatorversions crd: %w", err)
 	}
 
-	_, err = extensionsClientset.CustomResourceDefinitions().Get("instances.kudo.dev", v1.GetOptions{})
+	// create the clientset
+	kudoClientset, err := versioned.NewForConfig(config)
 	if err != nil {
-		// timeout details above for first CRD
-		if os.IsTimeout(err) {
-			return nil, err
-		}
-		return nil, fmt.Errorf("instances crd: %w", err)
+		return nil, err
```

Comment:
I think you need to handle the timeout here as you do in line 54 since this got moved

Reply:
No need for special handling, the `err` is just directly returned, the same that would happen with an extra check for timeout